src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -774,7 +774,9 @@ def post_init(self):
         super().post_init()
 
         if self.config.force_lm_head_in_fp32:
-            self.lm_head.to(torch.float32)
+            # FIXME @thomasw21: it's quite annoying that weight tie is not done in `super().post_init()` but in `from_pretrained`, not sure about the reason why. Consequently, we need to modify the word embeddings instead ...
+            # self.lm_head.to(torch.float32)
+            self.transformer.word_embeddings.to(torch.float32)
```

Comment:
Yeah I'm not sure why, but when instantiating a model using `from_pretrained` weight tying isn't done in `post_init` but rather in `from_pretrained`. This causes the old implementation to fail as you would `cast lm_head to fp32-> tie lm_head weight to word embeddings` which causes to ignore the fp32 casting. Ideally we'd want to do `tie lm_head weight to word embeddings -> cast lm_head to fp32` (which should also cast the word_embeddings). So I guess my question is why is weights tie done in `from_pretrained`?